### PR TITLE
feat(metric-stats): Add metric_stats generic metrics namespace

### PR DESCRIPTION
**Pull request by Dav1dde**
Adds the `metrics_stats` namespace and some initial strings.

All strings added are currently unused.

Depends on a librelay release after: https://github.com/getsentry/relay/pull/3267
Epic: https://github.com/getsentry/relay/issues/3147

### CODE REVIEW

File: src/sentry/sentry_metrics/indexer/strings.py [hunk 1]
```diff
@@ -189,6 +189,11 @@
     "is_application": PREFIX + 269,
     "platform": PREFIX + 270,
     "os.version": PREFIX + 271,
+    "mri": PREFIX + 272,
+    "mri.type": PREFIX + 273,
+    "mri.namespace": PREFIX + 274,
+    "outcome": PREFIX + 275,
```
Comment (Dav1dde) [inline src/sentry/sentry_metrics/indexer/strings.py:195]: `outcome` is already used, but only in `custom` metrics AND only by Sentry orgs

File: src/sentry/sentry_metrics/indexer/strings.py [hunk 2]
```diff
@@ -189,6 +189,11 @@
...
+    "reason": PREFIX + 276,
```
Comment (Dav1dde) [inline src/sentry/sentry_metrics/indexer/strings.py:196]: `reason` is already used, but only in `custom` metrics (which is still in alpha) AND for a single org in the `transactions` usecase with seemingly low volume

File: src/sentry/sentry_metrics/indexer/strings.py [hunk 3]
```diff
@@ -236,15 +242,22 @@
     "d:bundle_analysis/bundle_size@byte": PREFIX + 700,
 }
 
+# 800-899
+METRIC_STATS_METRIC_NAMES = {
+    "g:metric_stats/volume@none": PREFIX + 800,
```
Comment (jjbayer) [inline src/sentry/sentry_metrics/indexer/strings.py:247]: Isn't `volume` a counter rather than a gauge?

**Review by jjbayer — APPROVED** [review]
See question about counter vs. gauge.

File: src/sentry/sentry_metrics/indexer/strings.py [hunk 4]
```diff
@@ -191,6 +191,12 @@
     "os.version": PREFIX + 271,
     # Performance Score
     "sentry.score_profile_version": PREFIX + 272,
+    # Metric stats
+    "mri": PREFIX + 273,
```
[the remark this answers is not in the document]
Comment (Dav1dde) [reply src/sentry/sentry_metrics/indexer/strings.py:195]: I did checked all of them (see PR description)

**Review by iker-barriocanal — APPROVED** [review]
Does the PR require any follow-ups before the use case is used?

**Pull request merged**